models/Asset.php

```diff
@@ -158,6 +158,17 @@ protected function getBlockedVars(): array
 
         if (!$this->isInDumpState()) {
             // for caching asset
+
+            foreach($this->customSettings as $customSetting) {
```

Comment:
With this approach you cannot access the metadata when the item is loaded from the cache, right, because it's not getting refreshed on demand 🤔

Reply:
You are right. In our case this contained only garbage data but in general of course you MAY be interested in such data. Maybe we could check in `EmbeddedMetaDataTrait::getEmbeddedMetaData()` if any of the loaded fields has value `null` (or any other dummy value which we set in Asset::__sleep() for too long values) and reload metadata via `handleEmbeddedMetaData()` in this case?

Reply:
@brusch ?

Reply:
@BlackbitDevs Yep, I guess that would work 👍

Reply:
Have adjusted this PR to set `$this->customSettings['embeddedMetaDataExtracted'] = false;` if we stripped one of the big `embeddedMetaData` items. This way after deserializing the object meta data will get fetched again if `getEmbeddedMetaData()` gets called. So you can access ALL embedded metadata but you will not pollute Redis with hundreds of MB garbage data - which is expensive and would not help anything when loading the asset object because unserializing such a monster takes much longer than to reinitialize it from scratch.